stock_by_warehouse/models/product_product.py

```diff
@@ -0,0 +1,187 @@
+import json
+from collections import defaultdict
+from odoo import api, fields, models, _
+from odoo.tools import float_is_zero
+
+UNIT = 'Product Unit of Measure'
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    qty_available_not_res = fields.Float(
+        string='Qty Available Not Reserved',
+        digits=UNIT,
+        compute='_compute_qty_available_not_reserved',
+    )
+
+    warehouses_stock = fields.Text(
+        store=False, readonly=True,
+    )
+
+    warehouses_stock_location = fields.Text(store=False, readonly=True)
+
+    warehouses_stock_recompute = fields.Boolean(store=False)
+
+    @api.onchange('warehouses_stock_recompute')
+    def _warehouses_stock_recompute_onchange(self):
+        if not self.warehouses_stock_recompute:
+            self.warehouses_stock_recompute = True
+            return
+        self.warehouses_stock = self._compute_get_quantity_warehouses_json()
+        self.warehouses_stock_location = self._compute_get_stock_location()
+        self.warehouses_stock_recompute = True
+
+    def _product_available_not_res_hook(self, quants):
+        """Hook used to introduce possible variations"""
+        return False
+
+    def _prepare_domain_available_not_reserved(self):
+        domain_quant = [
+            ('product_id', 'in', self.ids)
+        ]
+        domain_quant_locations = self._get_domain_locations()[0]
+        domain_quant.extend(domain_quant_locations)
+        return domain_quant
+
+    @api.depends_context('warehouse', 'force_company')
```

Comment:
Delete  ``` force_company ``` https://github.com/odoo/odoo/blob/14.0/odoo/models.py#L5100

Reply:
thanks!, I replace it with company
https://github.com/odoo/odoo/blob/14.0/odoo/api.py#L226